### ISSUE TITLE
Check inventory space before selling

### ISSUE THREAD

**Issue by cyberkiller6** [ACEmulator/ACE#1281]
### Bug Report or Feature Request (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

### Repro steps.
Create a character
Use command /currency to get 100mmds
Go to a NPC, sell those 100mmds.
Instead of receiving the only pyreal you have pack space for, the NPC should not allow you to sell and display an error of some sort that say's you don't have enough pack space to sell all 100 mmds.

### Desired functionality.
Instead of receiving the only pyreal you have pack space for, the NPC should not allow you to sell and display an error of some sort that say's you don't have enough pack space to sell all 100 mmds.

**Comment by kdmxp256** [comment]
I would say the probably a player's encumbrance should be checked, also, to ensure that it doesn't go out of bounds.  Additional tasks that these checks should probably be performed with are crafting and emotes that have NPCs giving items to players.

**Comment by Mag-nus** [comment]
When you sell items, do they only go into your main pack or can they spill over into side packs as well?

**Comment by kdmxp256** [comment]
In buying items or gaining pyreals from selling, the latter option is the case, if there isn't any space available in the main pack.

**Comment by fartwhif** [comment]
are you sure?  i remember retail merchants not using the landscape packs whenever the main pack is full when buying or selling, there is a configuration option to always use your "currently open pack" so maybe you have that enabled?  IMHO it doesn't have to match retail and could probably be smarter than retail, which might even be easier to implement than perfectly cloning the retail server behavior.

**Comment by Mag-nus** [comment]
Should be resolved with #1452
